javalib/src/main/scala/java/io/CharArrayReader.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.
+ *  The ASF licenses this file to You under the Apache License, Version 2.0
+ *  (the "License"); you may not use this file except in compliance with
+ *  the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package java.io
+
+// Adapted from Apache Harmony CharArrayReader.java
+//   https://github.com/apache/harmony/blob/02970cb7227a335edd2c8457ebdde0195a735733/classlib/modules/luni/src/main/java/java/io/CharArrayReader.java
+
+class CharArrayReader extends Reader {
+  protected var buf: Array[Char] = null
+  protected var pos: Int = 0
+  protected var markedPos: Int = -1
+  protected var count: Int = 0
+
+  def this(buf: Array[Char]) {
+    this()
+    this.buf = buf
+    this.count = buf.length
+  }
+
+  def this(buf: Array[Char], offset: Int, length: Int) {
+    this()
+    /*
+     * The spec of this constructor is broken. In defining the legal values
```

Comment:
I don't understand how that's "broken". It's well specified in all possible cases. This comment does not seem helpful in blaming the spec.

Reply:
Removed the comments to avoid confusion, thanks!